cmd/nerdctl/image_prune_test.go

```diff
@@ -102,12 +102,15 @@ func TestImagePruneFilterUntil(t *testing.T) {
 	testutil.RequiresBuild(t)
 	testutil.RegisterBuildCacheCleanup(t)
 
-	// Docker image's created timestamp is set based on base image creation time.
-	testutil.DockerIncompatible(t)
-
 	base := testutil.NewBase(t)
+	// For deterministically testing the filter, set the image's created timestamp to 2 hours in the past.
+	base.Env = append(base.Env, fmt.Sprintf("SOURCE_DATE_EPOCH=%d", time.Now().Add(-2*time.Hour).Unix()))
+
 	imageName := testutil.Identifier(t)
-	t.Cleanup(func() { base.Cmd("rmi", "--force", imageName) })
+	t.Cleanup(func() {
+		// Image should have been pruned; but cleanup on failure.
+		base.Cmd("rmi", "--force", imageName).AssertOK()
```

Comment:
Run here, instead of AssertOK, is preferable.
Cleanup is not meant to test anything - nor fail.

Also, generally it is good hygiene to run the cleanup routine before the test as well - in case there is any leftovers from a previous borked run.